### ISSUE TITLE
Outdated history for pair and Unknown message type

### ISSUE THREAD

**Issue by Danson77** [freqtrade/freqtrade#8713]
Runing two bots on Raspberry Pi 4 4 RAM GB, I used both systems and added RTC just to see if it makes difference, its run as service  
  * Operating system: Ubuntu Server x64 and Raspberry Pi OS x64
  * Python Version: python3.11
  * CCXT version: 3.0.103
  * Freqtrade Version: 2023.5.dev
  
Im looking to fix the outdated ticks that keep appearing in log because bot not making any trades, I tried increasing "outdated_offset": 5 to like 20 to remove warnings but it still made no trades and some pairs where showing then 38-89 min outdated, I though the bot not refreshing fast enough or for too long and dont make it for pair selection so I changed "markets_refresh_interval": 5 from 30 but it makes no difreance, I want the Outdated history for pair to be updated every 5min,  its suspicious that all 75 pairlists i need got outdated, I tried enableRateLimit": true rate limit tried from 45 to 200, 1200 and 2400, I had bot trade few months ago without a problem and now I'm stuck with this logs and new config I rewrote thinking it would fix the error, also there is some kind of Unknown message type: analyzed_df , Unknown message type: new_candle and Unknown message type: whitelist also (httpcore - DEBUG - http11.receive_response_headers.failed exception=ReadTimeout(TimeoutError())) which could be my problem solution but I have no documentation on how to fix it, i hope I haven't missed any errors, I'm using binance and kucoin bot get same error but kucoin shows 429 which once in log on start maybe someone got a nice rate limit tested to share with because it seams like sometimes the 429 error don't show up with certain rate limit, but that's different issue. Could it be that Volume Pair list is doing this, I can try to test in static pairlist later 

Hers config i prefer to use but with outdated_offset 5
[config.txt](https://github.com/freqtrade/freqtrade/files/11601187/config.txt)

Produces this Log with verbosity 2
[freqtrade-2023-05-30_15-47-34.log](https://github.com/freqtrade/freqtrade/files/11601551/freqtrade-2023-05-30_15-47-34.log)

This is the logs by config I would prefer to use but with normal "markets_refresh_interval": 30 and "outdated_offset": 20 to get ride of outdated pairs waring so bot keeps pairs open for traid, all error accept Outdated once pair was 35 min old only one popped up, will this not hurt bot and let trade normaly or the data will be too old and bot will make poor decisions?

[binance.log](https://github.com/freqtrade/freqtrade/files/11604530/binance.log)
[kucoin.log](https://github.com/freqtrade/freqtrade/files/11604533/kucoin.log)

**Comment by xmatthias** [comment]
Well debug level logs are pretty much useless - it's way too verbose to see anything.

The answer to this is quite quick though - your strategy is too heavy / the system is too light / underpowered for this type of strategy.

Populating all signals takes about ~8s per pair. 
While this doesn't sound like too much on it's own - for 75 pairs, you'll come up with about 10 minutes of analysis time per pair (8 * 75 / 60).
As you're running on 5m candles, once the candle actually becomes available, it's outdated by ~2 candles.

The messages are therefore correct.
I also suspect that the 8s  are not the maximum it can take - i just picked one random entry at the end.
with 75 pairs, 1 more second per pair will make a difference of over 1 minute.


taking 2 lines as example, removing everything in the middle:

```
2023-05-30 15:56:52,456 - freqtrade.strategy.interface - DEBUG - Populating indicators for pair HIGH/USDT
...
2023-05-30 15:56:59,257 - freqtrade.strategy.interface - DEBUG - Populating indicators for pair LIT/USDT.
```

Simply put - either use a lighter strategy (without heavy processing, which this strategy is apparently doing), or a strategy which is better optimized for runtime speed, reduce your pairlist to something the bot can handle, or use a system with more power. 


---

Changing outdated_offset to anything bigger than the candle length WILL cause the bot to trade VERY poorly.

it'll act on signals that should've happened X candles ago (with 20 minutes offset, it's 4 5m candles). 
At that point, that signal is most likely either either no longer valid or price did the move already (and you're late buying at a bad rate).

**Comment by Danson77** [comment]
Thank you for feeedback, I prefer verbosity 0 too but I thought debug would show something more

The strategy is probably too heavy but I tested with powerful machine which would be my gaming pc and Outdated data pair was still showing up.

That calculation is pretty good explanation, each pair probably takes its sweet seconds which Warning actually doing its job

With outdated_offset I agree as well I dont want to trade with older then 5m candles 
but the strategy has those options which made me think it could be useful to trade with outdated candels but i belive it can lead to potential poor performance too 

# Optimal timeframe for the strategy.
    timeframe = '5m'
    info_timeframes = ['15m','1h','4h','1d']
# BTC informatives
    btc_info_timeframes = ['5m','15m','1h','4h','1d']  I still had it on 20 but I made a breakthrough
# Number of candles the strategy requires before producing valid signals
    startup_candle_count: int = 800

Now I made a breaktrough and binace bot made a trade but not the kucoin, it had still set 20 as offest for outdated but in binance log there is no outdated data for pair like in kucoin, All I did was cleaned cache and freqtrade keeping only some plots and configs on both rasberrys pies, installed again all dependecies, created new envirement, everything kind of fresh freqtrade and NostalgiaForInfinityX2 pulled from git, kucoin runs on ubuntu server and binance runs on rasbery pi os x 64 newest

Config Binance
[Binance.txt](https://github.com/freqtrade/freqtrade/files/11626010/Binance.txt)

Log Binance
[Binance.log](https://github.com/freqtrade/freqtrade/files/11625989/Binance.log)

Config Kucoin
[kucoin.txt](https://github.com/freqtrade/freqtrade/files/11626051/kucoin.txt)
Log Kucoin
[Kucoin.log](https://github.com/freqtrade/freqtrade/files/11625990/Kucoin.log)

Now the question is why kucoin has 429 error still and how to sort out rate limit error and Outdated history for pair XDB/USDT. Last tick is 31 to 46 minutes old but biance doesn't I made sure I created environment like on binance bot

My next step would be to set outdated_offset on binance and see if it throws error Outdated history which in my opinion it shouldnt because it has no Outdated history for pair XDB/USDT. Last tick is 31 to 46 minutes old warnings which to me it says the calculations are done fast enough

Kucoin enviremnt I will have to reset again maybe install rasberry pi os since its working better?

Later log!! I have now set 40 pair list and outdated_offset 5, no outdated pairs!!! Yeay!! But one thing worries me is that IMX/USDT is the best I found out to work with Nostalgiaforinfinityx2 and its not present in trading pair list in logs

**Comment by xmatthias** [comment]
Kucoins 429 error is actually a 429000 error.
That's a kucoin "speciality" - which they don't seem to get a hold of.
It's a "global, for all users" rate-limit per second, which you can hit at random times. There's nothing a client can do against this.
their support recommends to "retry immediately" - which we do through special handling (skips the backoff, as the message also states).

having outdated offset set to 20 means you won't get that message UNLESS it exceeds the 20 minutes - which i'd probably not recommend to use.

**Comment by Danson77** [comment]
Thank you for confirming information about 429 I will try to ignore it, as for the rest pairlist that have too many pairs will cause outdated data, both bots don't show that error no more if limited to 45 and 40 for Kucoin . Probably one last question, I used "dataformat_ohlcv": "hdf5" and "dataformat_trades": "hdf5" but I haven't seen many people use this format, would json or jsongz would be more preferably for trades format and improve performance ?

**Comment by xmatthias** [comment]
I did a benchmark on dataformats a while ago, which is [in the docs](https://www.freqtrade.io/en/latest/data-download/#dataformat-comparison).

in quick, hdf5 is better than json* - but feather beats them both by a long shot.
There's no good reason for parquet - but it was a 90% copy/paste from feather - and i didn't know it's performance before having it :laughing:

It should however be noted that the dataformat is only relevant for "data at rest".
a regular bot in trade mode won't touch (or need) that - so the format is only relevant for backtesting/hyperopt / download-data - and in trade mode IF you're using freqAI - which i don't think you are.

**Comment by Danson77** [comment]
@xmatthias Im actually using freqai but I turned it off for now because of the data required to be downloaded first time I start bot which takes time, so what would be a data format and settin in that case with this freqai parameters that I got from example config
 "freqai": {
    "enabled": false, 
    "purge_old_models": 2, 
    "train_period_days": 15, 
    "backtest_period_days": 7, 
    "fit_live_predictions_candles": 100, 
    "live_retrain_hours": 0.5,
    "identifier": "RL_model",

    "rl_config": {
      "train_cycles": 25, 
      "add_state_info": true, 
      "max_trade_duration_candles": 300,l
      "max_training_drawdown_pct": 0.02, 
      "cpu_count": 8,
      "model_type": "PPO", 
      "policy_type": "MlpPolicy", 
      "model_reward_parameters": {
        "rr": 1,
        "profit_aim": 0.025 
      }
    },
    "feature_parameters": {
      "include_timeframes": [ "5m", "15m", "1h", "4h", "1d" ], 
      "include_corr_pairlist": [
        "BTC/USDT:USDT",
        "ETH/USDT:USDT"
      ],
      "label_period_candles": 24, 
      "include_shifted_candles": 2, 
      "DI_threshold": 0.9, 
      "weight_factor": 0.9,
      "principal_component_analysis": false, 
      "use_SVM_to_remove_outliers": true, 
      "indicator_periods_candles": [ 10, 20 ],
      "plot_feature_importances": 0

    }
    "data_split_parameters": {
      "test_size": 0.25, 
      "random_state": 1 
    },
    "model_training_parameters": {
      "n_steps": 1024,
      "batch_size": 32 
    }
  },

I haven't yet tried feather format yet I will give it a try sounds like fun Thank you. Also side note Kucoin no matter the limit 40 or 45 still showing outdated pair or two that are 15 minutes so I think this is just Kucoin because binance working fine or the environment is broken

**Comment by xmatthias** [comment]
you'll need to keep in mind that with freqAI, computing power is even more important, as it has to do inference (and usually, parallel training).

As your features explode 5x timeframes * 2x corr pairlist * ...  ([total calculation is at the end of this section](https://www.freqtrade.io/en/stable/freqai-feature-engineering/#defining-the-features) - memory requirements might be high - as well as inference time (The model has a lot of features to evaluate / consider).

freqAI seems however to be disabled in the above config - so it's not really relevant, at least not with the config section shown above.
